apps/taskman/mixins.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+
+from .constants import JIRA_TASKMAN_AUTO_SYNC_FLAW
+
+
+class JiraTaskSyncMixin(models.Model):
+    """
+    mixin for syncing the model to the Jira
+
+    this mixin does not perform validation thus it should be
+    inherited after other mixins that performs it to ensire data
+    correctedness before syncing it
+    """
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, jira_token=None, **kwargs):
+        """
+        save the model by storing to Jira and then persisting in database
+
+        Jira sync is conditional based on environment variable so
+        development environment not enforces it
+        """
+        if JIRA_TASKMAN_AUTO_SYNC_FLAW:
```

Comment:
I am thinking about the condition here. It seems to me that even in a single environment with one value of `JIRA_TASKMAN_AUTO_SYNC_FLAW` we need to have two approaches here.

The reason is that we should not be creating the tasks for the historical flaws.

Probably for now we should not even create the tasks for flaws created originally in SFM2/Bugzilla but only for those created by OSIDB. If this makes sense to you I would suggest a similar behavior as it is in BBSync mixin in a way that you would additionally check the `jira_token` presence. The collectors would not provide it (so no task creation) and API queries which require the token would provide it. This would be a distinguisher of whether to create a task or not.

Reply:
Actually the third option is going to be a flaw draft created by the integrated Assembler. I think that this would behave similarly to the API call in the sense of providing the Jira token but of course the service one would be used as it is all automation.